### PR TITLE
Git conflicts validator, Fixes #36

### LINKED ISSUE #36: Suggestion: Check for merge conflicts
Sometimes, those pesky `<<<<<` can slip in a less expected way :). Example check:

```ruby
files.select { |f| File.read(f).match(/^<<<<<<<[ \t]/)
```

### PR DESCRIPTION
**Pull request by agwozdowski**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 - #21: Replace readme validator with required files validators, which allows us to
   define what files you want to require in your project
 - #35: Added gemfile validator which checks if Gemfile contains gems from local path
+- #36: Added git conflicts validator which checks if exist file with git conflict marker
```
Comment (mensfeld) [inline CHANGELOG.md]: Added git conflicts validator which checks if there are file with git conflict marker - sounds better imho

File: README.md [hunk 2]
```diff
@@ -68,6 +68,7 @@ determine, which you can use in your project:
 | rubycritic             | -         | Generates a RubyCritic for given project                                              |
 | allowed_extensions     | -         | Checks that all the files in ./config directory have an allowed extension             |
 | yml_parser             | -         | Checks that parameters of all yaml files in ./config directory have some value        |
+| git_conflicts          | -         | Checks if exist file with git conflict marker                                         |
```
Comment (mensfeld) [inline README.md]: Checks if there are files with git conflict marker

File: lib/polish_geeks/dev_tools/commands/git_conflicts.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+module PolishGeeks
+  module DevTools
+    module Commands
+      # Validator used to check if all files don't contain merge conflict
+      # @example: <<<<<<< HEAD
```
Comment (mensfeld) [inline lib/polish_geeks/dev_tools/commands/git_conflicts.rb]: what is <<<<<<<<<<< HEAD? ;)

File: lib/polish_geeks/dev_tools/commands/git_conflicts.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
+module PolishGeeks
+  module DevTools
+    module Commands
```
Comment (mensfeld) [inline lib/polish_geeks/dev_tools/commands/git_conflicts.rb]: @nijikon  wouldn't it be better to fgrep -R instead? It seams to me pretty heavy in terms of performance. Open each file, etc. What if there are huge binary files (images)? This version will use a looooooot of resources. I vote for fgrep -R

**Comment by agwozdowski** [comment]
I made the changes. @nijikon do we need checking on ruby 2.1.8? Because I have some problem on travis with that:
https://travis-ci.org/polishgeeks/polishgeeks-dev-tools/builds/107584293

**Comment by agwozdowski** [comment]
@nijikon green and rebased with master

**Comment by nijikon** [comment]
@agwozdowski Please rebase again.

**Comment by agwozdowski** [comment]
@nijikon rebased

**Comment by mensfeld** [comment]
@agwozdowski  is this still required? I believe bundle audit or brakeman has this functionality

**Comment by agwozdowski** [comment]
@mensfeld probably it's not required anymore. I'll close it.

**Pull request closed without merging**